numpy/lib/utils.py

```diff
@@ -193,7 +193,44 @@ def deprecate(*args, **kwargs):
     else:
         return _Deprecate(*args, **kwargs)
 
-deprecate_with_doc = lambda msg: _Deprecate(message=msg)
+def deprecate_with_doc(msg):
+    """
+    Returns new _Deprecate class object. 
+    The object can be used to Issue a DeprecationWarning, by passing `func`
+    as arguement,this adds warning to `old_name`'s docstring, rebinds 
+    ``old_name.__name__`` and returns the new function object.
+
+    This function may also be used as a decorator.
+
+    See Also
+    --------
+    deprecate`
+
+    Parameters
+    ----------
+    message : str
+    	    Additional explanation of the deprecation. Displayed in the 
+       	    docstring after the warning.
+
+    Returns
+    -------
+    _Deprecate object : object
+        The _Deprecate class object.
+
+    Examples
+    --------
+    Note that ``olduint`` returns a value after printing DeprecationWarning 
+    with msg:
+
+    >>>oldobj = np.deprecate_with_doc("Use np.int_ instead.")
+    >>>olduint = oldobj(np.uint)
+    >>>DeprecationWarning: `uint64` is deprecated! #may vary
+    ... Use np.int_ instead.
+    >>>olduint(6)
+    >>>6
```

Comment:
```suggestion
    >>> olduint(6)
    DeprecationWarning: `uint64` is deprecated!
        Use np.int_ instead.
    6
```
Two things:
* Only inputs are preceded with `>>>` (or `...`), outputs are not,
* The warning is only issued after calling `olduint`, not beforehand.